execution-engine/engine-tests/src/profiling/perf.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -e
+
+RED='\033[0;31m'
+CYAN='\033[0;36m'
+NO_COLOR='\033[0m'
+EE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/../../.." >/dev/null 2>&1 && pwd)"
+BASE_TEMP_DIR="/tmp/simple-transfer-perf"
+THIS_RUN_TEMP_DIR="${BASE_TEMP_DIR}/$RANDOM"
+
+check_for_perf() {
+  if ! [[ -x "$(command -v perf)" ]]; then
+    printf "${RED}perf not installed${NO_COLOR}\n\n"
+    printf "For Debian, try:\n"
+    printf "${CYAN}sudo apt install linux-tools-common linux-tools-generic linux-tools-$(uname -r)${NO_COLOR}\n\n"
+    printf "For Redhat, try:\n"
+    printf "${CYAN}sudo yum install perf${NO_COLOR}\n\n"
+    exit 127
+  fi
+}
+
+run_perf() {
+  cd $EE_DIR
+  make build-contracts
+  cd engine-tests/
+  cargo build --release --bin state-initializer
+  cargo build --release --bin simple-transfer
+  ../target/release/state-initializer --data-dir=../target | perf record -g --call-graph dwarf ../target/release/simple-transfer --data-dir=../target
+  mkdir -p ${THIS_RUN_TEMP_DIR}
+  mv perf.data ${THIS_RUN_TEMP_DIR}
+}
+
+check_or_clone_flamegraph() {
+  FLAMEGRAPH_DIR="${BASE_TEMP_DIR}/Flamegraph"
```

Comment:
```suggestion
  FLAMEGRAPH_DIR="${BASE_TEMP_DIR}/FlameGraph"
```

otherwise it will always reclone...

Reply:
I would be okay with just `/tmp/FlameGraph` as well

Reply:
(as it simplifies using `mktemp`)

Reply:
I agree with the nitpick for "g" to "G", but I don't think it would keep re-cloning since it adds `.../Flamegraph` to the path, so on the second run it would successfully execute the previously-cloned files.

Reply:
Ah good point.